presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/ApplyTableScanRedirection.java

```diff
@@ -97,6 +98,16 @@ public Result apply(TableScanNode scanNode, Captures captures, Context context)
                     if (destinationColumnHandle == null) {
                         throw new PrestoException(COLUMN_NOT_FOUND, format("Did not find handle for column %s in destination table %s", destinationColumn, destinationTable));
                     }
+
+                    // validate that redirected types match source types
+                    Type sourceType = context.getSymbolAllocator().getTypes().get(entry.getKey());
+                    Type redirectedType = metadata.getColumnMetadata(context.getSession(), destinationTableHandle.get(), destinationColumnHandle).getType();
+                    if (!sourceType.equals(redirectedType)) {
```

Comment:
Would here be a need in the future to support coercions? the type sets supported by connectors are not the same

Reply:
If your target catalog is Hive, then I don't think coercions would be needed as Hive supports most (all?) of Presto types with ORC format. For other targets, some coercion might be needed.

Reply:
> Hive supports most (all?) of Presto types with ORC format

eg `timestamp with time zone` or `json` are not supported